openerp/service/server.py

```diff
@@ -795,7 +795,6 @@ def process_work(self):
 
             import openerp.addons.base as base
             base.ir.ir_cron.ir_cron._acquire_job(db_name)
-            openerp.modules.registry.RegistryManager.delete(db_name)
 
```

Comment:
As I wrote in the other part, there should be no pending invalidation left on the registry.
IMO, the signaling should occur in method `_acquire_job` above, because that's were the registry is accessed.

That being said, we can avoid dropping the registry if it was in use.
I would simply delete the registry **if** it was not there before...